crates/aptos/src/move_tool/bytecode.rs

```diff
@@ -0,0 +1,306 @@
+        self.command.command_type = BytecodeCommandType::Decompile;
+        self.command.execute().await
+    }
+}
+
+impl BytecodeCommand {
+    async fn execute(self) -> CliTypedResult<String> {
+        let inputs = if let Some(path) = self.bytecode_path.clone() {
+            if self.package_path.is_some() {
+                return Err(CliError::CommandArgumentError(
+                    "Cannot provide both `--package-path` and `--bytecode-path".to_owned(),
+                ));
+            }
+            vec![path]
+        } else if let Some(path) = self.package_path.clone() {
+            read_dir_files(path.as_path(), |p| {
+                p.extension()
+                    .map(|s| s == MOVE_COMPILED_EXTENSION)
+                    .unwrap_or_default()
+            })?
+        } else {
+            return Err(CliError::CommandArgumentError(
+                "Must provide one of `--package-path` or `--bytecode-path".to_owned(),
+            ));
+        };
+
+        let mut report = vec![];
+        let mut last_out_dir = String::new();
+        for bytecode_path in inputs {
+            let bytecode_path = bytecode_path.as_path();
+            let extension = bytecode_path
+                .extension()
+                .context("Missing file extension for bytecode file")?;
+            if extension != MOVE_COMPILED_EXTENSION {
+                return Err(CliError::UnexpectedError(format!(
+                    "Bad source file extension {:?}; expected {}",
+                    extension, MOVE_COMPILED_EXTENSION
+                )));
+            }
+
+            let (output, extension) = match self.command_type {
+                BytecodeCommandType::Disassemble => {
+                    (self.disassemble(bytecode_path)?, DISASSEMBLER_EXTENSION)
+                },
+                BytecodeCommandType::Decompile => {
+                    (self.decompile(bytecode_path)?, DECOMPILER_EXTENSION)
+                },
+            };
+
+            let output_dir = if let Some(dir) = self.output_dir.clone() {
+                dir
+            } else {
+                bytecode_path.parent().expect("has parent dir").to_owned()
+            };
+            last_out_dir = output_dir.display().to_string();
+
+            let output_file = output_dir
+                .join(bytecode_path.file_name().expect("file name"))
+                .with_extension(extension);
+            check_if_file_exists(output_file.as_path(), self.prompt_options)?;
+
+            // Create the directory if it doesn't exist
+            create_dir_if_not_exist(output_dir.as_path())?;
+
+            // write to file
+            write_to_user_only_file(
+                output_file.as_path(),
+                &output_file.display().to_string(),
+                output.as_bytes(),
+            )?;
+            report.push(
+                output_file
+                    .file_name()
+                    .expect("file name")
+                    .to_string_lossy()
```

Comment:
we should use `display()` instead of `to_string_lossy()`

Reply:
I'd like to but `&OsStr` doesn't have it.